### DISCUSSION TITLE
account client and intiated action

### DISCUSSION THREAD

**Discussion by ahcamy2021** [keycloak/keycloak#10809] [Feedback]
Hi,

I noticed that from keycloak's version 13, the account client is available in React, but, you still offer some functionnalities in it using FreeMarker, I mean the functionnality 'Update Password', I understaood that you have FreeMarker because you used the concept of initiated action.

My question, do you have a vision in the next keycloak versions to migrate your flow's interfaces (Registration, authentification and reset Credentials) in react instead of Freemarker?

What do you advice to project that wants to add some functionnalities in account client already available as initiated action, use them or developp them in React to avoid rework in may be next version?

Thanks in advance,

**Comment by ssilvert** [comment]
I think you can expect some of these to be implemented with React.  We've been talking about that for registration recently.

For things that follow "application initiated action" flow, I think you can expect for that to continue.  But that doesn't mean it won't be implemented in React.  Decisions will be made on a case by case basis.

**Reply by aidachami-qc** [reply]
@ssilvert Thanks for you reply.
